consensus/propagation/types/unmarshalling_test.go

```diff
@@ -0,0 +1,90 @@
+package types
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	cmtrand "github.com/tendermint/tendermint/libs/rand"
+	"github.com/tendermint/tendermint/proto/tendermint/mempool"
+	"github.com/tendermint/tendermint/state"
+	"github.com/tendermint/tendermint/types"
+)
+
+// TestTxsToParts extensive testing of the txs to parts method
+// that recovers the parts from mempool txs.
+func TestTxsToParts(t *testing.T) {
```

Comment:
[could be a followup issue]
I like this test and have written similar style heavy weight random tests before. I think this is combining correctness testing and fuzzing however. It might actually be better if we have table driven tests for as many edge cases as we can think of, then have a fuzzer ontop

Reply:
https://github.com/celestiaorg/celestia-core/issues/1666